--- src/external/cctv.worker.js	2026-06-25 14:07:43.110224540 +0800
+++ src/external/cctv.worker.orig.js	2026-06-14 16:50:30.241110815 +0800
@@ -1,75 +1,7 @@
 /* 2026-05-12 18:20:40  4f63a66bf8e9c0c825f898b5c1015373 */
 console.debug('CCTV_Woker_BTime', '2026-05-12 18:20:40');
-export { CNTVModule };
 var CNTVModule = function() {
         var _scriptDir = "undefined" != typeof document && document.currentScript ? document.currentScript.src : void 0;
-
-        // a very simple xhr implementation to fit the need of __emscripten_fetch_xhr()
-        globalThis.XMLHttpRequest = class {
-            #url;
-            #method;
-            #resolveFunc;
-            onload;
-            onreadystatechange;
-            onerror;
-            response;
-            status = 0;
-            statusText = "";
-            readyState = 0;
-
-            open(method, url, async = true) {
-                this.#url = url;
-                this.#method = method;
-                if (this.#url === "https://tv.cctv.com/Library/H5player.json")
-                    this.#resolveFunc = this.#resolveH5PlayerJSON;
-                else
-                    this.#resolveFunc = fetch;
-
-                if (!async) // let's hope not so
-                    throw new Error("not supported");
-
-                this.readyState = 1;
-                this.onreadystatechange?.();
-            }
-
-            async #resolveH5PlayerJSON() {
-                // Blob has an interface similar to Response
-                return new Blob([
-                    JSON.stringify({
-                        h5player: {
-                            ver: 20190904,
-                            md5: "c7ed5a71dbe4dee1a2ba171f660ee98d",
-                            BTime: "2019-09-04-20:25:10"
-                        }
-                    })
-                ]);
-            }
-
-            send(body) {
-                this.#resolveFunc(
-                    this.#url,
-                    { body, method: this.#method }
-                ).then(resp => {
-                    this.readyState = 4;
-                    this.status = resp.status;
-                    this.statusText = resp.statusText;
-
-                    if (!resp.ok)
-                        throw new Error;
-
-                    return resp.arrayBuffer();
-                }).then(abuffer => {
-                    this.response = abuffer;
-
-                    this.onload?.();
-                    this.onreadystatechange?.();
-                }).catch(err => {
-                    this.onerror?.();
-                    this.onreadystatechange?.();
-                });
-            }
-        };
-
         return function(CNTVModule) {
             CNTVModule = CNTVModule || {};
             var Module = void 0 !== CNTVModule ? CNTVModule : {},
@@ -472,22 +404,8 @@
             };
             var ASM_CONSTS = [function($0) {
                 function theAnswer(thearg) {
-                    var i = {
-                        location: {
-                            hash: "",
-                            host: "",
-                            hostname: "",
-                            href: "blob:https://tv.cctv.com/5bca710b-9f02-41f0-a9f1-102bbc65192a",
-                            origin: "https://tv.cctv.com",
-                            pathname: "",
-                            port: "",
-                            protocol: "blob:",
-                            search: ""
-                        }
-                    };
-                    i.self = { location: i.location };
                     var name = UTF8ToString(thearg),
-                        a = eval(`i.${name}`),
+                        a = eval(name),
                         b = lengthBytesUTF8(a) + 1,
                         c = _malloc(b);
                     return stringToUTF8(a, c, b), c
@@ -2795,22 +2713,7 @@
             }
 
             function emval_get_global() {
-                var i = {
-                    location: {
-                        hash: "",
-                        host: "",
-                        hostname: "",
-                        href: "blob:https://tv.cctv.com/5bca710b-9f02-41f0-a9f1-102bbc65192a",
-                        origin: "https://tv.cctv.com",
-                        pathname: "",
-                        port: "",
-                        protocol: "blob:",
-                        search: ""
-                    }
-                };
-                i.self = { location: i.location };
-                return i;
-                // return "object" == typeof globalThis ? globalThis : Function("return this")()
+                return "object" == typeof globalThis ? globalThis : Function("return this")()
             }
 
             function __emval_get_global(A) {
@@ -2901,7 +2804,7 @@
             }
 
             function _emscripten_get_callstack(A, e, t) {
-                A = "blob:https://tv.cctv.com/5bca710b-9f02-41f0-a9f1-102bbc65192a";
+                A = _emscripten_get_callstack_js(A);
                 return !e || t <= 0 ? lengthBytesUTF8(A) + 1 : stringToUTF8(A, e, t) + 1
             }
 
@@ -3262,9 +3165,6 @@
                     w = !!(4 & c),
                     u = !!(32 & c),
                     c = !!(16 & c);
-                __emscripten_fetch_xhr(A, n, E, I, o);
-                return A;
-
                 if ((!c || "EM_IDB_STORE" === t || "EM_IDB_DELETE" === t) && !Fetch.dbInstance) return E(A), 0;
                 if ("EM_IDB_STORE" === t) {
                     var d = HEAPU32[e + 84 >> 2];
